src/sage/matroids/chow_ring.py

```diff
@@ -0,0 +1,346 @@
+
+..MATH::
+
+    A(M)_R := R[y_{e_1}, \ldots, y_{e_n}, x_{F_1}, \ldots, x_{F_k}] / (I_M + J_M)
+
+    where
+
+..MATH::
+
+    (I_M + J_M)
+
+    is the augmented Chow ring ideal of matroid `M` of Feitchner-Yuzvinsky
+    presentation.
+
+The augmented Chow ring ideal of atom-free presentation is the quotient ring:
+
+..MATH::
+
+    A(M)_R := R[x_{F_1}, \ldots, x_{F_k}] / I_{af}M
+
+    where
+
+    I_{af}M
+    
+    is the augmented Chow ring ideal of matroid `M` of atom-free presentation.
+
+.. SEEALSO::
+
+    :mod: sage.matroids.chow_ring_ideal
+
+REFERENCES:
+
+    - [FY2004]_
+    - [AHK2015]_
+"""
+
+from sage.matroids.chow_ring_ideal import ChowRingIdeal_nonaug, AugmentedChowRingIdeal_fy, AugmentedChowRingIdeal_atom_free
+from sage.rings.quotient_ring import QuotientRing_generic
+from sage.categories.graded_algebras_with_basis import GradedAlgebrasWithBasis
+from sage.categories.commutative_rings import CommutativeRings
+from sage.misc.misc_c import prod
+from itertools import product
+
+class ChowRing(QuotientRing_generic):
+    r"""
+    The class of Chow ring, a multi-polynomial quotient ring.
+
+    INPUT:
+
+
+    - `M` -- a matroid.
+    - `R` -- a ring.
+    - ``augmented`` -- a Boolean value. When ``True``, it returns the augmented Chow
+      ring. If ``False``, it returns the Chow ring
+
+    OUTPUT: Chow ring of matroid `M`.
+
+    EXAMPLES::
+
+        sage: from sage.matroids.chow_ring import ChowRing
+
+        sage: M1 = matroids.catalog.P8pp()
+        sage: ch = ChowRing(M=M1, R=QQ, augmented=False)
+        sage: ch
+        Chow ring of P8'': Matroid of rank 4 on 8 elements with 8 nonspanning circuits
+    """
+    def __init__(self, R, M, augmented, presentation=None):
+        r"""
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.chow_ring import ChowRing
+
+            sage: I = ChowRing(M=matroids.Wheel(3), R=QQ, augmented=False)
+            sage: TestSuite(I).run()
+        """
+        self._matroid = M
+        self._augmented = augmented
+        self._presentation = presentation
+        if augmented:
+            if presentation=='fy':
+                self._ideal = AugmentedChowRingIdeal_fy(M, R)
+            elif presentation=='atom-free':
+                self._ideal = AugmentedChowRingIdeal_atom_free(M, R)
+        else:
+            self._ideal = ChowRingIdeal_nonaug(M, R) #check method to get ring
+        C = CommutativeRings().Quotients() & GradedAlgebrasWithBasis(R).FiniteDimensional()
+        QuotientRing_generic.__init__(self, R=self._ideal.ring(), I=self._ideal, names=self._ideal.ring().variable_names(), category=C)
+
+    def _repr_(self):
+        r"""
+        EXAMPLE::
+
+        sage: from sage.matroids.chow_ring import ChowRing
+
+        sage: M1 = matroids.catalog.Fano()
+        sage: ch = ChowRing(M=M1, R=QQ, augmented=False)
+        sage: ch
+        Chow ring of Fano: Binary matroid of rank 3 on 7 elements, type (3, 0)
```

Comment:
```suggestion
        EXAMPLES::

            sage: M1 = matroids.catalog.Fano()
            sage: ch = M1.chow_ring(R=QQ, augmented=False)
            sage: ch
            Chow ring of Fano: Binary matroid of rank 3 on 7 elements, type (3, 0)
```
Many parts of this change also need to be duplicated below.